globals_test.go

```diff
@@ -299,6 +299,46 @@ func TestLoadGlobals(t *testing.T) {
 				),
 			},
 		},
+		{
+			name:   "stack with globals referencing globals using try",
+			layout: []string{"s:stack"},
+			globals: []globalsBlock{
+				{
+					path: "/stack",
+					add: globals(
+						str("field", "value"),
+						expr("field_try", `try(global.field, "error")`),
+						expr("undefined_try", `try(global.undefined, "undefined")`),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					str("field", "value"),
+					str("field_try", "value"),
+					str("undefined_try", "undefined"),
+				),
+			},
+		},
+		{
+			name:   "stack with globals referencing metadata using try",
+			layout: []string{"s:stack"},
+			globals: []globalsBlock{
+				{
+					path: "/stack",
+					add: globals(
+						expr("path_try", `try(terramate.path, "error")`),
+						expr("undefined_try", `try(terramate.undefined, "undefined")`),
```

Comment:
that's not how try works in terraform ;)

Reply:
Indeed, when I looked into the docs I saw this

"try evaluates all of its argument expressions in turn and returns the result of the first one that does not produce any errors."

Imagined that reference to undefined would just be an error.

But missed this:

"The try function will not catch errors relating to constructs that are probably invalid even before dynamic expression evaluation, such as a malformed reference or a reference to a top-level object that has not been declared"

Sorry for the noise

Reply:
all good.. we can actually implement it the way we want.... for now i think it's better to carp out if a variable is not defined at all on any level... wdyt?